### ISSUE TITLE
Incorporate Design System Alert Component - Health & Benefits

### ISSUE THREAD

**Issue by TimRoe** [department-of-veterans-affairs/va-mobile-app#8467]
## Description 
As a veteran, I want to have a consistent visual experience interacting with the VA between web and mobile.

As part of this story we want to:   
- Upgrade the Health & Benefits areas of the app to use the design system Alert component
- Reach out to the design system team if encountering issues or to provide other feedback if appropriate

## Acceptance Criteria
  

-  [ ] If not already done within the app, update `@department-of-veterans-affairs/mobile-component-library` package to version 0.15.0 or higher so the Alert component is available
-  [ ] If not already done within the app, update `@department-of-veterans-affairs/eslint-config-mobile` package to version 0.14.0 and run `yarn nodes` to install it so linting will issue warnings for outdated Alert components
-  [ ] With prior AC complete, run `yarn lint` and you'll get a list in the console of Alertbox and CollapsibleAlert components that need to be updated
   -  [ ] Parse through the list of 33 Alertbox warnings and update all within Health & Benefits code to use the design system Alert
   - [ ] Parse through the list of 5 CollapsibleAlert warnings and update all within Health & Benefits code to use the design system Alert

**Accessibility Requirements**
- [ ] Validate Alert component accessibility functionality works as expected

## Notes & Open Questions
- Does this require QA? Yes
    
Dev Notes: 
- The design system Alert component opted to not include haptics or scroll-to/focus behavior as not inherently relevant to the component itself for all mobile use cases
   - Code-wise, it is expected haptics should be capable of being handled at the app-level since it is a non-visual action and can be added alongside the Alert
   - Code-wise, it was unclear if the scroll-to/focus behavior can be fully mimicked without integration into the component--evaluate as part of this ticket and reach out to the design system team to determine a solution if it cannot adequately meet expectations

## Ticket Checklist

- [ ] Linked to an Epic

**Comment by TimRoe** [comment]
Hello @ala-yna , giving a heads up that this ticket was created for incorporating the Alert component into the app from the Health & Benefits side. I added all details I thought relevant, but may have missed something not being familiar with H&B team processes so feel free to update as you see fit to fit it into the roadmap.

**Comment by ala-yna** [comment]
Thank you Tim! Is there an Epic associated with this work?

**Comment by TimRoe** [comment]
> Is there an Epic associated with this work?

There was and Kelly added it to it when I mentioned creating the ticket to her. Looks to be:
> [department-of-veterans-affairs/va-mobile-app#7620](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7620) Incorporate Alert Design System Component into the app - Health & Benefits

**Comment by Sparowhawk** [comment]
waiting on this one until appts feature-xl and the SM help panel tickets are merged

**Comment by alexandec** [comment]
For QA, changes are in the following locations:

- src/components/EncourageUpdate.tsx
- src/components/MessageAlert.tsx
- src/components/WaygateWrapper.tsx
- src/components/WhatsNew.tsx
- src/components/CommonErrorComponents/CallHelpCenter.tsx
- src/components/CommonErrorComponents/DowntimeError.tsx
- src/components/CommonErrorComponents/ErrorAlert.tsx
- src/screens/auth/LoginScreen/LoginScreen.tsx
- src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/SelectFile.tsx
- src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/TakePhotos.tsx
- src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/UploadOrAddPhotos/UploadOrAddPhotos.tsx
- src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimTimeline/ClaimTimeline.tsx
- src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/EstimatedDecisionDate/EstimatedDecisionDate.tsx
- src/screens/BenefitsScreen/ClaimsScreen/ClaimsHistoryScreen/ClaimsHistoryScreen.tsx
- src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.tsx
- src/screens/HealthScreen/Appointments/Appointments.tsx
- src/screens/HealthScreen/CernerAlert/CernerAlert.tsx
- src/screens/HealthScreen/Pharmacy/PrescriptionDetails/PrescriptionsDetailsBanner.tsx
- src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistory.tsx
- src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistoryNoPrescriptions.tsx
- src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistoryNotAuthorized.tsx
- src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.tsx
- src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.tsx
- src/screens/HealthScreen/SecureMessaging/CernerAlertSM/CernerAlertSM.tsx
- src/screens/HealthScreen/SecureMessaging/EditDraft/EditDraft.tsx
- src/screens/HealthScreen/SecureMessaging/ReplyMessage/ReplyMessage.tsx
- src/screens/HealthScreen/SecureMessaging/StartNewMessage/StartNewMessage.tsx
- src/screens/HealthScreen/SecureMessaging/StartNewMessage/Attachments/Attachments.tsx
- src/screens/HealthScreen/SecureMessaging/ViewMessage/IndividualMessageErrorComponent.tsx
- src/screens/HealthScreen/SecureMessaging/ViewMessage/ViewMessageScreen.tsx
- src/screens/HealthScreen/Vaccines/NoVaccineRecords/NoVaccineRecords.tsx
- src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/AddressValidation/AddressValidation.tsx
- src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/EditAddressScreen/EditAddressScreen.tsx
- src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/EditEmailScreen/EditEmailScreen.tsx
- src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/EditPhoneNumberScreen/EditPhoneNumberScreen.tsx
- src/screens/HomeScreen/ProfileScreen/SettingsScreen/NotificationsSettingsScreen/NotificationsSettingsScreen.tsx
- src/screens/PaymentsScreen/DirectDepositScreen/EditDirectDepositScreen/EditDirectDepositScreen.tsx
- src/screens/PaymentsScreen/DirectDepositScreen/HowToUpdateDirectDepositScreen/HowToUpdateDirectDepositScreen.tsx
- src/screens/PaymentsScreen/PaymentHistory/NoPayments/NoPaymentsScreen.tsx

**Comment by rbontrager** [comment]
Waiting on the increased spacing ticket for the alert component to be resolved before I approve this ticket.

**Comment by alexandec** [comment]
@rbontrager increased spacing is now added so this should be all set for QA

**Comment by rbontrager** [comment]
@alexandec Found a couple issues with the spacing.  I think its an app code issue and not a component issue considering most of the alerts look fine but I may be wrong.  

It looks like there is too much spacing between the header and the text in the following places:
- Appointments cerner alert (src/screens/HealthScreen/CernerAlert/CernerAlert.tsx)
- Messages cerner alert (src/screens/HealthScreen/SecureMessaging/CernerAlertSM/CernerAlertSM.tsx)
- Screen reader accessibility alert in take photos (src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/TakePhotos/TakePhotos.tsx)
- Mailing/Home Verify your address screen (src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/AddressValidation/AddressValidation.tsx)

It also looks like the spacing increased between the first and second paragraphs in the following places:
-  VA Mobile app not working right now
-  No Vaccine Records (src/screens/HealthScreen/Vaccines/NoVaccineRecords/NoVaccineRecords.tsx)

Finally there are couple instances where the spacing between the description text and the button or the description text and tty is too big:
- What's New (src/components/WhatsNew.tsx)
- Waygates (src/components/WaygateWrapper.tsx)

**Comment by rbontrager** [comment]
Verified that the alert component follows the design in the storybook given by the design system.  Verified that the new alert component works like it previously did from an a11y perspective.  Verified that all issues with spacing in my comment above were fixed and that all alerts given in the list by Chris have been updated to use the new alert component.  Approved by QA.
